litestar/plugins/problem_details.py

```diff
@@ -0,0 +1,120 @@
+
+        return problem_details_exc.to_response(req)
+
+    return _exception_handler
+
+
+def _http_exception_to_problem_detail_exception(exc: HTTPException) -> ProblemDetailsException:
+    return ProblemDetailsException(status_code=exc.status_code, detail=exc.detail, extra=exc.extra, headers=exc.headers)
+
+
+class ProblemDetailsException(HTTPException):
+    """A problem details exception as per RFC 9457."""
+
+    _PROBLEM_DETAIL_MEDIA_TYPE = "application/problem+json"
+
+    def __init__(
+        self,
+        *args: Any,
+        detail: str = "",
+        status_code: int | None = None,
+        headers: dict[str, str] | None = None,
+        extra: dict[str, Any] | list[Any] | None = None,
+        type_: str = "about:blank",
+        title: str | None = None,
+        instance: str | None = None,
+    ) -> None:
+        """Initialize ``ProblemDetailsException``.
+
+        Args:
+            *args: if ``detail`` kwarg not provided, first arg should be error detail.
+            detail: Exception details or message. Will default to args[0] if not provided.
+            status_code: Exception HTTP status code.
+            headers: Headers to set on the response.
+            extra: An extra mapping to attach to the exception.
+            type_: The type field in the problem details.
+            title: The title field in the problem details.
+            instance: The instance field in the problem details.
+        """
+
+        super().__init__(*args, detail=detail, status_code=status_code, headers=headers, extra=extra)
+
+        self.type_ = type_
+        self.title = title
+        self.instance = instance
+
+    def to_response(self, request: Request) -> Response:
+        """Convert the problem details exception into a ``Response.``"""
+
+        problem_details = {"type": self.type_, "status": self.status_code}
+        for attr in ("title", "instance", "detail", "extra"):
+            if (value := getattr(self, attr, None)) is not None:
+                problem_details[attr] = value
+
+        return Response(problem_details, headers=self.headers, media_type=self._PROBLEM_DETAIL_MEDIA_TYPE)
+
+
+@dataclass
+class ProblemDetailsConfig:
+    """The configuration object for ``ProblemDetailsPlugin.``"""
+
+    exception_handler: ProblemDetailsExceptionHandlerType = _problem_details_exception_handler
+    """The exception handler used for ``ProblemdetailsException.``"""
+
+    enable_for_all_http_exceptions: bool = False
+    """Flag indicating whether to convert all exceptions into ``ProblemDetailsException.``"""
+
+    exception_to_problem_detail_map: ExceptionToProblemDetailMapType = field(default_factory=dict)
+    """A mapping to convert exceptions into ``ProblemDetailsException.``
+
+    All exceptions provided in this will get a custom exception handler where these exceptions
+    are converted into ``ProblemDetailException`` before handling them.
+    """
```

Comment:
Adding HTTPException to the map currently takes precedence over `enable_for_all_http_exceptions` - should that be documented?

Reply:
Oh good point. Yeah, I did that intentionally so that they could override the `HTTPException` conversion into problem details if needed.